content/blog/how-to-backup-and-restore-stateful-applications-on-kubernetes-using-kasten-k10-in-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,660 @@
+
+![](/img/k10-dashboard-backup.png)
+
+Y﻿ou can also check the **Data Usage** page to see the data used by database backups:
+
+![](/img/k10-data-backup.png)
+
+I﻿n the cluster, after snapshot of the MySQL database, you can check there is a *VolumeSnapshot* *'k10-csi-snap-ltxzrwxgp6r5pwkp'* created f﻿rom the source PVC *'mysql-pvc'* in the namespace *mysql*, together with a *VolumeSnapshotContent* object created at cluster level. The *READYTOUSE* of the *VolumeSnapshot* should be showing as *true*:
+
+```shell
+$ kubectl get volumesnapshot -n mysql
+NAME                            READYTOUSE   SOURCEPVC   SOURCESNAPSHOTCONTENT   RESTORESIZE   SNAPSHOTCLASS              SNAPSHOTCONTENT                                    CREATIONTIME   AGE
+k10-csi-snap-ltxzrwxgp6r5pwkp   true         mysql-pvc                           1Gi           gl-sbp-frank-gl1-sstor01   snapcontent-f3890356-d47f-4b36-a7e4-eb4c5792ec59   6d12h          6d12h
+
+ $ kubectl get volumesnapshotcontents
+NAME                                               READYTOUSE   RESTORESIZE   DELETIONPOLICY   DRIVER        VOLUMESNAPSHOTCLASS        VOLUMESNAPSHOT                  VOLUMESNAPSHOTNAMESPACE   AGE
+snapcontent-f3890356-d47f-4b36-a7e4-eb4c5792ec59   true         1073741824    Delete           csi.hpe.com   gl-sbp-frank-gl1-sstor01   k10-csi-snap-ltxzrwxgp6r5pwkp   mysql                     6d12h
+```
+
+T﻿his volume snapshot can be used for MySQL database restore.
+
+### Restore MySQL database
+
+B﻿efore showing the database restore, I﻿ will first delete some table from MySQL database to simulate a loss of data. Then, I will perform the database recovery using the Kasten K10.
+
+**1. Delete table**
+
+D﻿elete data from the table '*departments'* by typing the following commands:
+
+```shell
+$ mysql -h 127.0.0.1 -uroot -pCfeDemo@123 -P 42281 -Demployees
+Reading table information for completion of table and column names
+You can turn off this feature to get a quicker startup with -A
+
+Welcome to the MariaDB monitor.  Commands end with ; or \g.
+Your MySQL connection id is 15
+Server version: 5.6.51 MySQL Community Server (GPL)
+
+Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.
+
+Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
+
+MySQL [employees]> show tables;
++----------------------+
+| Tables_in_employees  |
++----------------------+
+| current_dept_emp     |
+| departments          |
+| dept_emp             |
+| dept_emp_latest_date |
+| dept_manager         |
+| employees            |
+| salaries             |
+| titles               |
++----------------------+
+8 rows in set (0,237 sec)
+
+MySQL [employees]> delete from departments;
+Query OK, 9 rows affected (1,523 sec)
+```
+
+I﻿f re-run the testing script *test_employees_sha.sql*, it will show the failures of *CRC* and *count* which indicate the loss of data in the MySQL database:
```

Comment:
Modify to "If you re-run the testing script..."
Place a comma after "count" 
If you rerun the testing script... it will show the failures of CRC and count, which indicate the loss of data...